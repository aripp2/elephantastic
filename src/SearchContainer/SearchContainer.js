import React from 'react';
import SearchForm from '../SearchForm/SearchForm';
import './SearchContainer.scss';

const SearchContainer = () => {
  return (
    <div>
      <SearchForm />
      <section>
        Choose a breed to see images. 
      </section>
    </div>
  )
}

export default SearchContainer;