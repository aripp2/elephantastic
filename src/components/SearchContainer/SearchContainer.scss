@import './src/variables.scss';

.searchImgs {
  height: 100vh;
  width: 80%;
  padding: 20px;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
}

.search-img {
  height: auto; 
  width: auto; 
  max-width: 45%; 
  max-height: 45%;
  border-radius: 5px;
}