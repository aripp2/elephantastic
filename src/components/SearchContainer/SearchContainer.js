import React from 'react';
import './SearchContainer.scss';
import PropTypes from 'prop-types';


const SearchContainer = ({ breedImages }) => {
  const selectedBreedImages = breedImages.map(image => {
    return <img
      className='search-img' 
      key={image.id}
      id={image.id}
      src={image.url}
      alt='dog'
    />
  })
  return (
      <section className='searchImgs'>
        {breedImages.length === 0 && <h2 className='selectPrompt'>Select a breed from the left to see images.</h2>}
        { selectedBreedImages } 
      </section>
  )
}

export default SearchContainer;

SearchContainer.propTypes = {
  breedImages: PropTypes.array
}