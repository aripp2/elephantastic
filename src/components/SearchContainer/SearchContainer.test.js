import React from 'react';
import { shallow } from 'enzyme';
import SearchContainer from './SearchContainer';

describe('SearchContainer', () => {
  let wrapper;
  const mockBreedImages = [
    {
      breeds: [{}],
      height: 853,
      id: "BbMFS3bU-",
      url: "https://cdn2.thedogapi.com/images/BbMFS3bU-.jpg",
      width: 1280
    },
    {
      breeds: [{}],
      height: 900,
      id: "8ogZCsDcp",
      url: "https://cdn2.thedogapi.com/images/8ogZCsDcp.jpg",
      width: 1200
    }];

  it('should match the snapshot', () => {
    wrapper = shallow(<SearchContainer breedImages={mockBreedImages} />)
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a message if breedImages is empty', () => {
    const mockEmpty = [];
    wrapper = shallow(<SearchContainer breedImages={mockEmpty} />)
    expect(wrapper).toMatchSnapshot();
  });

});