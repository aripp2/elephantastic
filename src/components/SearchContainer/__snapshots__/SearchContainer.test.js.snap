// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContainer should match the snapshot 1`] = `
<section
  className="searchImgs"
>
  <img
    alt="dog"
    className="search-img"
    id="BbMFS3bU-"
    key="BbMFS3bU-"
    src="https://cdn2.thedogapi.com/images/BbMFS3bU-.jpg"
  />
  <img
    alt="dog"
    className="search-img"
    id="8ogZCsDcp"
    key="8ogZCsDcp"
    src="https://cdn2.thedogapi.com/images/8ogZCsDcp.jpg"
  />
</section>
`;

exports[`SearchContainer should render a message if breedImages is empty 1`] = `
<section
  className="searchImgs"
>
  <h2
    className="selectPrompt"
  >
    Select a breed from the left to see images.
  </h2>
</section>
`;
